From 25bc2cbf504003f7f5ce73835cb1415889ec0c55 Mon Sep 17 00:00:00 2001
From: Fabian Ruff <fabian@progra.de>
Date: Wed, 30 Sep 2015 20:37:50 +0200
Subject: [PATCH 1/2] Use c.UserName when calling swift

---
 swift.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/github.com/ncw/swift/swift.go b/github.com/ncw/swift/swift.go
index 003e105..d31f1eb 100644
--- a/github.com/ncw/swift/swift.go
+++ b/github.com/ncw/swift/swift.go
@@ -463,7 +463,7 @@ func (c *Connection) Call(targetUrl string, p RequestOpts) (resp *http.Response,
 				req.Header.Add(k, v)
 			}
 		}
-		req.Header.Add("User-Agent", DefaultUserAgent)
+		req.Header.Add("User-Agent", c.UserAgent)
 		req.Header.Add("X-Auth-Token", authToken)
 		resp, err = c.doTimeoutRequest(timer, req)
 		if err != nil {

From 4a9dcde138f73b1bc433fa448d5b8094b7fc2725 Mon Sep 17 00:00:00 2001
From: Fabian Ruff <fabian.ruff@sap.com>
Date: Wed, 30 Sep 2015 21:40:47 +0200
Subject: [PATCH 2/2] use custom User-Agent for v2 and v3 auth requests

---
 auth.go    | 1 +
 auth_v3.go | 1 +
 2 files changed, 2 insertions(+)

diff --git a/github.com/ncw/swift/auth.go b/github.com/ncw/swift/auth.go
index ca35d23..3948bc1 100644
--- a/github.com/ncw/swift/auth.go
+++ b/github.com/ncw/swift/auth.go
@@ -156,6 +156,7 @@ func (auth *v2Auth) Request(c *Connection) (*http.Request, error) {
 		return nil, err
 	}
 	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", c.UserAgent)
 	return req, nil
 }
 
diff --git a/github.com/ncw/swift/auth_v3.go b/github.com/ncw/swift/auth_v3.go
index 7c375d2..b657747 100644
--- a/github.com/ncw/swift/auth_v3.go
+++ b/github.com/ncw/swift/auth_v3.go
@@ -177,6 +177,7 @@ func (auth *v3Auth) Request(c *Connection) (*http.Request, error) {
 		return nil, err
 	}
 	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", c.UserAgent)
 	return req, nil
 }
 
